PLAN-fix-mcp-search.md

```
@@ -0,0 +1,289 @@
+**File:** `src/services/worker/SearchManager.ts`
+
+Currently (around line 184–356):
+
+- A local `let chromaFailed = false` is set in the catch at line 304, but the error itself is discarded except for the `logger.warn` log line.
+- When `totalResults === 0 && chromaFailed`, line 356 returns the static lying string.
+
+Change shape (do not copy verbatim — match existing types):
+
+1. At line 184, also declare `let chromaFailureReason: { message: string; isConnectionError: boolean } | null = null;`.
+2. In the catch at line 301-304, populate it from the caught error:
+   ```ts
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)